Extensions/Xtensive.Orm.Reprocessing/Internals/ExecuteConfiguration.cs

```diff
@@ -37,14 +33,33 @@ public IExecuteConfiguration WithTransactionOpenMode(TransactionOpenMode transac
       return this;
     }
 
+    public IExecuteConfiguration WithExternalSession(Session session)
```

Comment:
What do you think about renaming this to just `WithSession` ?

Reply:
Well, I wanted the name to be as clear as possible. That's why I named it ```.WithExternalSession()```. I believe we can rename WithXXX methods but in ```ExecutionContext``` I'd prefer the name which underlines nature of the instance, maybe keep ```ExternalSession``` or replace it with ```UserSession``` (because we don't open it) or something.